src/app/ui/TaskBoard.tsx

```diff
@@ -123,25 +53,28 @@ export const TaskBoard = () => {
           overflowX: 'auto',
         }}
       >
-        {Object.entries(lists).map(([listId, listData], index) => (
-          <DragDropHandler
-            key={listId}
-            accept={'taskCard'}
-            index={index}
-            id={Number(listId)}
-            onDropItem={(item) => onDropItem(Number(listId), item)}
-          >
-            <TaskColumn key={listId} columnName={`List ${listId}`} taskCount={String(listData.length)}>
-              {listData.map((task: Task, index: number) => (
-                <DragDropHandler key={task.id} accept={'taskCard'} index={index} id={Number(listId)} moveCard={moveCard}>
-                  <Box onClick={() => router.push('/detail/WEB-01/my-new-task')}>
-                    <TaskCard assignee={task.assignee} key={task.id} />
-                  </Box>
-                </DragDropHandler>
-              ))}
-            </TaskColumn>
-          </DragDropHandler>
-        ))}
+        {workflowStates.map((list, index) => {
+          return (
+            <DragDropHandler key={list.id} accept={'taskCard'} index={index} id={list.id} onDropItem={onDropItem}>
+              <TaskColumn key={list.id} columnName={list.name} taskCount={calculateTaskCountBasedOnWorkflowStateId(list.id)}>
+                {filterTaskWithWorkflowStateId(list.id).map((task, index) => {
+                  return (
+                    <DragDropHandler key={task.id} accept={'taskCard'} index={index} id={task.id || ''}>
+                      <Box
+                        onClick={() => router.push(`/detail/${task.id}/${encodeToParamString(task.title || '')}/iu`)}
```

Comment:
Why are we using `task.title` as a dynamic param? Shouldn't `task.id` be enough? Task title will be fetched from the API anyways.

Reply:
In the server rendered apps, we try to get rid of the loading states as much as possible by fetching the data in the server and pre-rendering the parts of the page whenever possible. 
This is also an approach on pre-rendering the `title` in the details page even before the data is fetched. By passing the `title` through params in the server component ([link to code](https://github.com/copilot-platforms/tasks-app/blob/063f90c465e8b03d3f05b314cff12a11ddda8ec2/src/app/detail/%5Btask_id%5D/%5Btask_name%5D/%5Buser_type%5D/page.tsx#L20)), we can pre-render this part and show the users with something. This is an approach where we take benefits of server side rendering by displaying parts of pages without data fetching is completed. It is also a composition pattern for designing apps using mixed components. 

Beside this, I think putting `title` along with `id` as part of a route param is a tradition. For instance, this is an example of the linear details page route: `https://linear.app/copilotplatforms/issue/OUT-249/implement-remaining-task-actions`

Reply:
I see that's what Linear does as well.. although it seems weird at first because of the long URLs... in a later PR we could make it so that links like `tasks.app/WEB-01` redirects to `tasks.app/WEB-01/Do-something` as well (like Linear). Right now the `WEB-01` part is hardcoded so it might not be possible.

Reply:
There shouldn't be much issues if it is only used this way. However there might be a case where I have opened "tasks.app/WEB-01/Do-something" in my browser and the author of the ticket renames it from "Do something" to "Do something else".

I that case can you confirm if "Do something else" is shown as title when "tasks.app/WEB-01/Do-something" is opened? (Even if "Do-something" is used to set the initial title, the title is updated in place if necessary?

Let me know if this is confusing :bow:

Reply:
Good catch. The synchronization in the case title is changed was not implemented. Considering this and also considering the long URL until we have shorter task id like `WEB-01`, I have removed the task name from route param in d69c36b75be57aa82c305ceefd656dec1264f264 commit. 

In the future, we can have the title in route param by managing the synchronization in the case if title is changed, if required. Otherwise, this is fine. @rrojan